src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2535,21 +2550,46 @@ class CreateViewStatementSegment(BaseSegment):
 
 
 @ansi_dialect.segment()
-class DropStatementSegment(BaseSegment):
-    """A `DROP` statement."""
+class DropTableStatementSegment(BaseSegment):
+    """A `DROP TABLE` statement."""
+
+    type = "drop_table_statement"
 
-    type = "drop_statement"
-    # DROP {TABLE | VIEW} <Table name> [IF EXISTS} {RESTRICT | CASCADE}
     match_grammar = Sequence(
         "DROP",
-        OneOf(
-            "TABLE",
-            "VIEW",
-            "USER",
-        ),
+        "TABLE",
         Ref("IfExistsGrammar", optional=True),
         Ref("TableReferenceSegment"),
-        OneOf("RESTRICT", Ref.keyword("CASCADE", optional=True), optional=True),
+        Ref("DropBehaviorGrammar", optional=True),
+    )
+
+
+@ansi_dialect.segment()
+class DropViewStatementSegment(BaseSegment):
+    """A `DROP VIEW` statement."""
+
+    type = "drop_view_statement"
+
+    match_grammar = Sequence(
+        "DROP",
+        "VIEW",
+        Ref("IfExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        Ref("DropBehaviorGrammar", optional=True),
+    )
+
+
+@ansi_dialect.segment()
+class DropUserStatementSegment(BaseSegment):
+    """A `DROP` statement for role, user, or group."""
+
+    type = "drop_user_statement"
+
+    match_grammar = Sequence(
+        "DROP",
+        OneOf("ROLE", "USER", "GROUP"),
+        Ref("IfExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
     )
```

Comment:
sorry missed this before: split out into drop role, drop user, and drop group statements

Reply:
addressed in my PR